src/org/openbravo/common/hooks/PrintControllerHookManager.java

```diff
@@ -0,0 +1,124 @@
+   *
+   * @param hooks
+   *     the hooks to be sorted
+   * @return a list of hooks sorted by priority
+   */
+  public static List<PrintControllerHook> sortHooksByPriority(Instance<PrintControllerHook> hooks) {
+    List<PrintControllerHook> hookList = new ArrayList<>();
+    hooks.forEach(hookList::add);
+
+    hookList.sort((Comparator<Object>) (o1, o2) -> {
+      int o1Priority = (o1 instanceof PrintControllerHookPrioritizer) ? ((PrintControllerHookPrioritizer) o1).getPriority() : 100;
+      int o2Priority = (o2 instanceof PrintControllerHookPrioritizer) ? ((PrintControllerHookPrioritizer) o2).getPriority() : 100;
+
+      return Integer.compare(o1Priority, o2Priority);
+    });
+
+    return hookList;
+  }
+
+  /**
+   * Handles errors encountered during hook execution, recording the error message in the JSON parameters.
+   *
+   * @param jsonParams
+   *     the JSON parameters being processed
+   * @param isPreProcess
+   *     whether the error occurred during preProcess
+   * @param e
+   *     the exception that was thrown
+   * @param hook
+   *     the hook that encountered the error
+   * @throws JSONException
+   *     if there is an error processing the JSON parameters
+   */
+  private static void handleHookError(JSONObject jsonParams, boolean isPreProcess, Exception e,
+      PrintControllerHook hook) throws JSONException {
+    if (jsonParams.getBoolean(CANCELLATION)) {
+      throw new OBException(e.getMessage());
+    }
+    jsonParams.getJSONObject(RESULTS).put(FAILURES, true);
+    HashBasedTable<String, Boolean, String> messageInfo = (HashBasedTable<String, Boolean, String>) jsonParams.getJSONObject(
+        RESULTS).get(MESSAGE);
+    messageInfo.put(hook.getClass().getSimpleName(), isPreProcess, e.getMessage());
+    jsonParams.getJSONObject(RESULTS).put(MESSAGE, messageInfo);
+  }
+
+  /**
+   * Executes the hooks based on the specified method name (preProcess or postProcess).
+   * If any hook returns an error message, the execution stops and the error message is returned.
+   *
+   * @param jsonParams
+   *     the JSON parameters to be processed by the hooks
+   * @param methodName
+   *     the name of the method to be executed (preProcess or postProcess)
+   * @throws JSONException
+   *     if there is an error processing the JSON parameters
+   */
+  public void executeHooks(JSONObject jsonParams, String methodName) throws JSONException {
+    List<PrintControllerHook> hookList = sortHooksByPriority(hooks);
+    for (PrintControllerHook hook : hookList) {
+      if (StringUtils.equals(methodName, PREPROCESS)) {
+        try {
+          hook.preProcess(jsonParams);
+        } catch (Exception e) {
+          handleHookError(jsonParams, true, e, hook);
+        }
+      } else if (StringUtils.equals(methodName, POSTPROCESS)) {
+        try {
+          hook.postProcess(jsonParams);
+        } catch (Exception e) {
+          handleHookError(jsonParams, false, e, hook);
```

Comment:
If you used an else if statement to allow another futures method names the boolean parameter will be unnecessary.